app/services/task-assignment.js

```diff
@@ -29,16 +29,25 @@ export default Service.extend({
 
   _create(user, task) {
     let store = get(this, 'store');
-    return store.createRecord('user-task', { user, task })
-                .save();
+    let userTask = store.createRecord('user-task', { user, task });
+    return userTask.save().catch(() => {
+      userTask.deleteRecord();
```

Comment:
if an error occurs while creating a record, why make an API call to delete it vs just unloading it?